pkg/noderesourcetopology/common-plugin-helpers.go

```diff
@@ -0,0 +1,140 @@
+	topoclientset "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/clientset/versioned"
+	topologyinformers "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/informers/externalversions"
+	v1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/informers/externalversions/topology/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+)
+
+// no no mock only indexer
+type mockIndexer struct {
+	nodeTopologies nodeTopologyMap
+}
+
+func (m mockIndexer) ByIndex(indexName, indexKey string) ([]interface{}, error)      { return nil, nil }
+func (m mockIndexer) AddIndexers(cache.Indexers) error                               { return nil }
+func (m mockIndexer) GetIndexers() cache.Indexers                                    { return nil }
+func (m mockIndexer) ListIndexFuncValues(indexName string) []string                  { return nil }
+func (m mockIndexer) Index(indexName string, obj interface{}) ([]interface{}, error) { return nil, nil }
+func (m mockIndexer) IndexKeys(indexName, indexedValue string) ([]string, error)     { return nil, nil }
+
+func (m mockIndexer) Add(interface{}) error               { return nil }
+func (m mockIndexer) Update(obj interface{}) error        { return nil }
+func (m mockIndexer) Delete(obj interface{}) error        { return nil }
+func (m mockIndexer) List() []interface{}                 { return nil }
+func (m mockIndexer) ListKeys() []string                  { return nil }
+func (m mockIndexer) Replace([]interface{}, string) error { return nil }
+func (m mockIndexer) Resync() error                       { return nil }
+func (m mockIndexer) Get(obj interface{}) (item interface{}, exists bool, err error) {
+	return nil, false, nil
+}
+
+func (m mockIndexer) GetByKey(key string) (item interface{}, exists bool, err error) {
+	if v, ok := m.nodeTopologies[key]; ok {
+		return &v, ok, nil
+	}
+	return nil, false, fmt.Errorf("Node topology is not found: %v", key)
+}
+
+func findNodeTopology(nodeName string, data *commonPluginsData) *topologyv1alpha1.NodeResourceTopology {
+	klog.V(5).Infof("namespaces: %s", data.namespaces)
+	for _, namespace := range data.namespaces {
+		// NodeTopology couldn't be placed in several namespaces simultaneously
+		nodeTopology, err := data.lister.NodeResourceTopologies(namespace).Get(nodeName)
+		if err != nil {
+			klog.V(5).Infof("Cannot get NodeTopologies from NodeResourceTopologyNamespaceLister: %v", err)
+			continue
+		}
+		if nodeTopology != nil {
+			return nodeTopology
+		}
+	}
+	return nil
+}
+
+func initNodeTopologyInformer(masterOverride, kubeConfigPath string) (v1alpha1.NodeResourceTopologyInformer, error) {
+	kubeConfig, err := clientcmd.BuildConfigFromFlags(masterOverride, kubeConfigPath)
```

Comment:
Honestly, I like the idea of factoring the general code out.
I would like to see here singleton instead of creation a copy of kubeConfig, topoConfig, informerFactory (https://golang.org/pkg/sync/#Once), and just return a reference to instantiated once a time object of type v1alpha1.NodeResourceTopologyInformer